pilot/pkg/status/resource.go

```diff
@@ -102,7 +102,7 @@ func GetStatusManipulator(in any) (out Manipulator) {
 	if ret, ok := in.(*networking.ServiceEntryStatus); ok && ret != nil {
 		return &ServiceEntryGenerationProvider{ret}
 	}
-	return &NopStatusManipulator{in}
+	return &NoopStatusManipulator{in}
```

Comment:
**nit:** Nop is valid IMO, it's valid ASM for a lot of ISAs and used other places in the codebase. That being said we do also have noop so there's no standard and I don't feel the need to enforce any specific standard.

Reply:
Sure, assembly instructions typically have the convention of using three-letter mnemonics which dates back to the 50s. But I'm referring to the Istio codebase. But I checked again and you're right, there's at least 1 other `Nop` use case here:  
`type NopTester struct{}`.

Deciding to have no standards or naming conventions is certainly a choice. This was just me doing a drive-by refactor with something that I don't personally see often (I also don't work with assembly instructions and this was more of a personal preference). I didn't block John's PR because of it and I'm not opposed to leaving it as is. TBF, the shorter version, `Nop` does align more with idiomatic Go then `Noop` does.

You have a valid point!

Reply:
A valid concern here is that Nop in and of itself is a misnomer since this Nop struct actually does have some ops which produce a result...

It would be fair to expect the Nop type to drop your `any` right onto the metaphorical floor when you SetInner and to always return `nil` when you Unwrap but it in fact does not do that!